Import translations from babel and cleveref

eltos opened the pull request:
Imports all available translations from [babel](https://ctan.org/tex-archive/macros/latex/required/babel/base) and [cleveref](https://ctan.org/tex-archive/macros/latex/contrib/cleveref).

Closes #6581 by following up on [this comment](https://github.com/typst/typst/pull/6519#issuecomment-3018270925):
> This makes me think that we should probably have done that from the start. Just researching the LaTeX name data once and supporting all the languages they support immediately instead of having the terms decided by contributors one by one. I think that came up once in the past, but nobody ever followed up on it...

The PR has multiple commits:
- 35d6a1f1a41ab960a7b06819f575d903017b29a3 imports all translations, overwriting those that already existed in typst
- 717ba1a8eddff9841a8e2af6bd7d634705b22146 reverts the overwritten terms, providing a clean diff which could be reviewed by a [native speaker](https://github.com/typst/typst/commits/main/crates/typst-library/translations)
  ```diff
  - term used by babel/cleveref
  + term used by typst as of today
  ```
- 8595bb9936a64c57a1aa0625d89b1b295c44d142 adds the translations to `lang.rs` 
- 4c27710b8aa957f84b3dd0ef78d152c07b47ce9a adds language constants and fixes a wrong assignment of Filipino/Tagalog

Overall, the PR **does not change** any of the existing translations, **only adds** missing terms and locals ones.

## Import script

I used the following python script to parse babel and cleveref sources, extract the localized terms and generate typst translations. Since these contain many more terms than currently used by typst (some including abbreviations and plurals), I post the script here in case it might be useful in the future:

[babel_cleveref_to_typst.py](https://github.com/user-attachments/files/22680355/babel_cleveref_to_typst.py)

## Notes  
- Not all translations are complete in babel/cleveref, I have also imported incomplete translations if there is at least one term translated
- Capitalization of terms frequently differs between babel/cleveref and typst (`page` vs `Page`), as well as between locals. As stated above, existing terms are kept as-is, including capitalization. New terms are imported as-is, with whatever capitalization is used by babel/cleveref for the respective locale.
- For some locals, cleveref provides abbreviations (`Fig.` vs. `Figure`), but I chose to import the form babel provides. I noticed some inconsistent use of long-form vs. abbreviation in babel as well as typst translations. My feeling is, that this is more a question of editor style, but I don't know if some languages have a strong preference for either, a question which only a native speaker can answer, so I just kept them as-is.
- Both babel and cleveref are licensed under the [LaTeX Project Public License](https://www.latex-project.org/lppl.txt). ~I *believe* there is no problem, as this is not a derived work and "[The LPPL] license places no restrictions on works that are unrelated to the Work".~ **EDIT:** A notice was added to the notice file. However, I am not qualified to make a legal statement about the use of the extracted terms under typst's Apache-2.0 license. 

## Future

The following follow-up features might be useful, but are out of scope for this PR:
- Add singular/plural terms and natively combine multiple adjacent reference in typst using the plural forms, simmilar to how multiple adjacent citations are combined
- Provide a function to get translated terms in the different forms (long/abbreviation, singular/plural, capitalized/in-text) for use by templates or packages, such as https://typst.app/universe/package/smartaref

laurmaedje commented:
> I believe there is no problem, as this is not a derived work

Is it not? But regardless, it should be good if we just add an entry to the NOTICE file. The hyphenation patterns Typst uses are also LPPL and LPPL and Apache-2 are compatible.

eltos commented:
> add an entry to the NOTICE file

Done

laurmaedje commented:
I'd like to land this in 0.14, but https://github.com/typst/typst/pull/6619 introduced a few new terms and resulted in a few conflicts. If you have time to take a look, that would be appreciated! Otherwise, I'd do it myself.

I also plan to do rebase & merge to keep the individual commits as they are very useful for posterity, so a force push would be ideal.

eltos commented:
I have rebased onto main and added "footnote" translations while keeping the commit structure.
The terms "email" and "telephone" are unfortunately neither available in babel nor cleveref.

Andrew15-5 commented:
Are all [ISO 639-1/2/3 language codes](https://en.wikipedia.org/wiki/ISO_639) were included in the `Lang` impl? Does it add something useful if most of them are unused (more than 200, IIUC)?
